### PR TITLE
feat!: add more support for queries in gnomad_vcf_to_protein 

### LINKED ISSUE #362: Add more support for deletions/insertions `gnomad_vcf_to_protein`
Insertions where the reference has length of one and deletions where the alteration has a length of one are currently supported. We should expand the kinds of variants we can support for this method

### PR DESCRIPTION
**Pull request by korikuzma**
close #362 

@wesleygoar still in draft, but your feedback would be nice

### CODE REVIEW

**Comment by wesleygoar** [comment]
Screenshot 2024-01-11 at 12 48 10 PM
Example from vcf in varhouse

**Comment by wesleygoar** [comment]
image (12)
annotated by varhouse as...

**Review by wesleygoar — APPROVED** [review]
This all seems correct to me.

**Comment by wesleygoar** [comment]
@korikuzma we should consider asking @ahwagner to review as well to confirm.

**Comment by korikuzma** [comment]
Since this endpoint needs gene-normalizer data, I'm wondering if we should use ProteinSequenceConsequence. Right now, I have the catvar pydantic models in metakb (it made development easier). Maybe we should move them here or in a different place?

File: variation/schemas/gnomad_vcf_to_protein_schema.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from pydantic import StrictStr
+
+from variation.schemas.normalize_response_schema import NormalizeService
+
+
+class GnomadVcfToProteinService(NormalizeService):
+    """Define response for gnomad vcf to protein service"""
+
+    gene_context: Optional[core_models.Gene] = None
+    vrs_ref_allele_seq: Optional[StrictStr] = None
```
Comment (korikuzma) [inline variation/schemas/gnomad_vcf_to_protein_schema.py:14]: I could have just added this to variation extensions field, but thought it'd be easier for varcat if we had a field to retrieve

**Comment by wesleygoar** [comment]
> Since this endpoint needs gene-normalizer data, I'm wondering if we should use ProteinSequenceConsequence. Right now, I have the catvar pydantic models in metakb (it made development easier). Maybe we should move them here or in a different place?

Hmm. I will default to where @ahwagner thinks is best for them since he runs the whole show.

File: variation/schemas/gnomad_vcf_to_protein_schema.py (hunk 1 above)
Comment (wesleygoar) [reply variation/schemas/gnomad_vcf_to_protein_schema.py:14]: I am always pro making varcat better (easier to develop). @ahwagner any objections?

**Review by wesleygoar — APPROVED** [review]

File: tests/test_gnomad_vcf_to_protein.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+        },
+        "state": {"type": "LiteralSequenceExpression", "sequence": "PS"},
+    }
+    return models.Allele(**params)
+
+
+@pytest.fixture(scope="module")
+def multi_nuc_sub_neg():
+    """Create test fixture for substitution with more than 1 nucleotide change on the
+    negative strand (CA1139661942)
```
Comment (ahwagner) [inline tests/test_gnomad_vcf_to_protein.py:204]:
```suggestion
    """Create test fixture for the protein consequence of a multinucleotide substitution on the
    negative strand (CA1139661942)
```

File: tests/test_gnomad_vcf_to_protein.py [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+        "location": braf_600loc,
+        "state": {"sequence": "V", "type": "LiteralSequenceExpression"},
+        "type": "Allele",
+    }
+    return models.Allele(**params)
+
+
+@pytest.fixture(scope="module")
+def kras_g12d():
+    """Fixture for KRAS G12C"""
```
Comment (ahwagner) [inline tests/test_gnomad_vcf_to_protein.py:159]:
```suggestion
    """Fixture for KRAS G12D"""
```

File: tests/test_gnomad_vcf_to_protein.py [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+            "end": 236,
+        },
+        "state": {"type": "LiteralSequenceExpression", "sequence": "G"},
+    }
+    return models.Allele(**params)
+
+
+@pytest.fixture(scope="module")
+def delins_pos():
+    """Create test fixture for delins on positive strand (CA645561524)"""
```
Comment (ahwagner) [inline tests/test_gnomad_vcf_to_protein.py:226]:
```suggestion
    """Create test fixture for the protein consequence of a delins on positive strand (CA645561524)"""
```

File: tests/test_gnomad_vcf_to_protein.py [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+            "end": 752,
+        },
+        "state": {"type": "LiteralSequenceExpression", "sequence": "Q"},
+    }
+    return models.Allele(**params)
+
+
+@pytest.fixture(scope="module")
+def delins_neg():
+    """Create test fixture for delins on negative strand (ClinVar ID 1217291)"""
```
Comment (ahwagner) [inline tests/test_gnomad_vcf_to_protein.py:247]:
```suggestion
    """Create test fixture for the protein consequence of a delins on negative strand (ClinVar ID 1217291)"""
```

File: tests/test_gnomad_vcf_to_protein.py [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+    assertion_checks(resp, atad3a_i7t)
+    assert resp.warnings == []
+
+    # Reading Frame 3, Positive Strand
+    resp = await test_handler.gnomad_vcf_to_protein("1-1512289-T-G")
+    assertion_checks(resp, atad3a_i7m)
+    assert resp.warnings == []
+
+    resp = await test_handler.gnomad_vcf_to_protein("12-25245350-C-T")
+    assertion_checks(resp, kras_g12d)
```
Comment (ahwagner) [inline tests/test_gnomad_vcf_to_protein.py:319]: Spot-checked this one against gnomAD. Looks good!

**Review by ahwagner — APPROVED** [review]
Overall outstanding work. 👍 
I noticed a few minor docstring clarifications and thread comments.

I spot tested a few of the provided tests, they check out.

**Comment by wesleygoar** [comment]
@korikuzma I am pretty sure I have reviewed this at least once if not more. Is there any reason for me to review again?

**Comment by korikuzma** [comment]
> @korikuzma I am pretty sure I have reviewed this at least once if not more. Is there any reason for me to review again?

@wesleygoar no, when changes get made the reviews are dismissed. I only re-request reviews when I need things to be reviewed again.

**Pull request merged**